### PR TITLE
Add the example 02 - BoW generator

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2257

Here, this is a BoW model generator implemented using Ballerina I/O operations. More information about the BoW model:
- https://machinelearningmastery.com/gentle-introduction-bag-words-model/
- https://en.wikipedia.org/wiki/Bag-of-words_model

Note, we cannot build this example here since it uses the regex library as a dependency, and it will create a circular dependency if we automatically run it during the build phase. Also, we use gRPC spec with certain modified as the sample raw text dataset.

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: examples/bow/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+```sh
+$ cd examples/bow
+$ bal run
+```
+
+Then provide the input value. This input value indicates the number of most frequent words users wants to retrieve. For example:
+
+```
+> Enter the frequency boundry value: 10
```
Comment (MadhukaHarith92) [inline examples/bow/README.md:29]: Is `frequency boundry value` a term specific to the BoW model? If not can we say like `Enter the number of most frequent words to retrieve: `

Comment (BuddhiWathsala) [reply examples/bow/README.md]: Sounds good. Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/447/commits/7c5853ff449370f3c48ff599eb374370e158ac86

File: examples/bow/main.bal [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        string[] tokens = regex:split(line, "\\s+");
+        foreach string token in tokens {
+            if token != "" {
+                string lowerCaseToken = token.toLowerAscii();
+                int? frequency = bow[lowerCaseToken];
+                if frequency is int {
+                    bow[lowerCaseToken] = frequency + 1;
+                } else {
+                    bow[lowerCaseToken] = 1;
+                }
```
Comment (MadhukaHarith92) [inline examples/bow/main.bal:37]:
```suggestion
                bow[lowerCaseToken] = frequency is int ? frequency + 1 : 1;
```

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**